src/main/java/com/faforever/client/mod/ModVaultController.java

```diff
@@ -289,6 +326,12 @@ private void appendSearchResult(List<ModVersion> modVersions, Pane pane) {
           return controller;
         }).collect(Collectors.toList());
 
+    Platform.runLater(() -> {
+      pane.getChildren().clear();
+    });
+
+    JavaFxUtil.assertBackgroundThread();
```

Comment:
if this is really the background thread why load fxml on the background thread just above :thinking:

Reply:
Also I guess make assert on method top